File: GitToolBox/src/main/java/zielu/gittoolbox/config/DateType.java
package zielu.gittoolbox.config;

import com.intellij.util.xmlb.annotations.Transient;
import zielu.gittoolbox.ResBundle;

public enum DateType {
  AUTO("date.type.auto"),
  RELATIVE("date.type.relative"),
  ABSOLUTE("date.type.absolute")
  ;

  private final String descriptionKey;

  DateType(String descriptionKey) {
    this.descriptionKey = descriptionKey;
  }

  @Transient
  public String getDescription() {
    return ResBundle.message(descriptionKey);
  }
}

File: GitToolBox/src/main/kotlin/zielu/gittoolbox/config/DateType.kt
package zielu.gittoolbox.config

import com.intellij.util.xmlb.annotations.Transient
import zielu.gittoolbox.ResBundle

internal enum class DateType(private val labelKey: String) {
  AUTO("date.type.auto"),
  RELATIVE("date.type.relative"),
  ABSOLUTE("date.type.absolute");

  @Transient
  fun getDisplayLabel() = ResBundle.message(labelKey)
}
